File: sdks/python/bolt_prediction_sdk/models/warning_strategy_config_schema.py
"""
WarningStrategyConfigSchema 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class WarningStrategyConfigSchema(SDKBaseModel):
    """预警策略配置"""

    strategy_type: int = Field(description="策略类型: 1=应报尽报, 2=精准报警")
    strategy_1_confidence_threshold: Optional[float] = Field(description="策略1置信度阈值", default=0.7)
    strategy_1_false_positive_threshold: Optional[float] = Field(description="策略1误报率阈值", default=0.05)
    strategy_2_confidence_threshold: Optional[float] = Field(description="策略2置信度阈值", default=0.95)
    strategy_2_false_negative_threshold: Optional[float] = Field(description="策略2漏报率阈值", default=0.1)

File: sdks/java/src/main/java/com/boltprediction/sdk/model/WarningStrategyConfigSchema.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 预警策略配置 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class WarningStrategyConfigSchema {

    @JsonProperty("strategy_type")
    private Integer strategyType;

    @JsonProperty("strategy_1_confidence_threshold")
    private Double strategy1ConfidenceThreshold;

    @JsonProperty("strategy_1_false_positive_threshold")
    private Double strategy1FalsePositiveThreshold;

    @JsonProperty("strategy_2_confidence_threshold")
    private Double strategy2ConfidenceThreshold;

    @JsonProperty("strategy_2_false_negative_threshold")
    private Double strategy2FalseNegativeThreshold;

    public Integer getStrategyType() {
        return strategyType;
    }

    public void setStrategyType(Integer strategyType) {
        this.strategyType = strategyType;
    }

    public Double getStrategy1ConfidenceThreshold() {
        return strategy1ConfidenceThreshold;
    }

    public void setStrategy1ConfidenceThreshold(Double strategy1ConfidenceThreshold) {
        this.strategy1ConfidenceThreshold = strategy1ConfidenceThreshold;
    }

    public Double getStrategy1FalsePositiveThreshold() {
        return strategy1FalsePositiveThreshold;
    }

    public void setStrategy1FalsePositiveThreshold(Double strategy1FalsePositiveThreshold) {
        this.strategy1FalsePositiveThreshold = strategy1FalsePositiveThreshold;
    }

    public Double getStrategy2ConfidenceThreshold() {
        return strategy2ConfidenceThreshold;
    }

    public void setStrategy2ConfidenceThreshold(Double strategy2ConfidenceThreshold) {
        this.strategy2ConfidenceThreshold = strategy2ConfidenceThreshold;
    }

    public Double getStrategy2FalseNegativeThreshold() {
        return strategy2FalseNegativeThreshold;
    }

    public void setStrategy2FalseNegativeThreshold(Double strategy2FalseNegativeThreshold) {
        this.strategy2FalseNegativeThreshold = strategy2FalseNegativeThreshold;
    }

}